core/ledger/kvledger/snapshot_test.go

```diff
@@ -0,0 +1,379 @@
+	conf, cleanup := testConfig(t)
+	defer cleanup()
+	snapshotRootDir := conf.SnapshotsConfig.SnapshotsRootDir
+	provider := testutilNewProviderWithCollectionConfig(
+		t,
+		"ns",
+		map[string]uint64{"coll": 0},
+		conf,
+	)
+	defer provider.Close()
+
+	// add the genesis block and generate the snapshot
+	blkGenerator, genesisBlk := testutil.NewBlockGenerator(t, "testLedgerid", false)
+	lgr, err := provider.Create(genesisBlk)
+	require.NoError(t, err)
+	defer lgr.Close()
+	kvlgr := lgr.(*kvLedger)
+	require.NoError(t, kvlgr.generateSnapshot())
+	verifySnapshotOutput(t,
+		snapshotRootDir,
+		kvlgr.ledgerID,
+		1,
+		protoutil.BlockHeaderHash(genesisBlk.Header),
+		kvlgr.commitHash,
+		"txids.data", "txids.metadata",
+	)
+
+	// add block-1 only with public state data and generate the snapshot
+	blockAndPvtdata1 := prepareNextBlockForTest(t, kvlgr, blkGenerator, "SimulateForBlk1",
+		map[string]string{"key1": "value1.1", "key2": "value2.1", "key3": "value3.1"},
+		nil,
+	)
+	assert.NoError(t, kvlgr.CommitLegacy(blockAndPvtdata1, &ledger.CommitOptions{}))
+	require.NoError(t, kvlgr.generateSnapshot())
+	verifySnapshotOutput(t,
+		snapshotRootDir,
+		kvlgr.ledgerID,
+		2,
+		protoutil.BlockHeaderHash(blockAndPvtdata1.Block.Header),
+		kvlgr.commitHash,
+		"txids.data", "txids.metadata",
+		"public_state.data", "public_state.metadata",
+	)
+
+	// add block-2 only with public and private data and generate the snapshot
+	blockAndPvtdata2 := prepareNextBlockForTest(t, kvlgr, blkGenerator, "SimulateForBlk2",
+		map[string]string{"key1": "value1.2", "key2": "value2.2", "key3": "value3.2"},
+		map[string]string{"key1": "pvtValue1.2", "key2": "pvtValue2.2", "key3": "pvtValue3.2"},
+	)
+	assert.NoError(t, kvlgr.CommitLegacy(blockAndPvtdata2, &ledger.CommitOptions{}))
+	require.NoError(t, kvlgr.generateSnapshot())
+	verifySnapshotOutput(t,
+		snapshotRootDir,
+		kvlgr.ledgerID,
+		3,
+		protoutil.BlockHeaderHash(blockAndPvtdata2.Block.Header),
+		kvlgr.commitHash,
+		"txids.data", "txids.metadata",
+		"public_state.data", "public_state.metadata",
+		"private_state_hashes.data", "private_state_hashes.metadata",
+	)
+
+	// add dummy entry in collection config history and commit block-3 and generate the snapshot
+	addDummyEntryInCollectionConfigHistory(t, provider, kvlgr.ledgerID)
+	blockAndPvtdata3 := prepareNextBlockForTest(t, kvlgr, blkGenerator, "SimulateForBlk3",
+		map[string]string{"key1": "value1.3", "key2": "value2.3", "key3": "value3.3"},
+		nil,
+	)
+	assert.NoError(t, kvlgr.CommitLegacy(blockAndPvtdata3, &ledger.CommitOptions{}))
+	require.NoError(t, kvlgr.generateSnapshot())
+	verifySnapshotOutput(t,
+		snapshotRootDir,
+		kvlgr.ledgerID,
+		4,
+		protoutil.BlockHeaderHash(blockAndPvtdata3.Block.Header),
+		kvlgr.commitHash,
+		"txids.data", "txids.metadata",
+		"public_state.data", "public_state.metadata",
+		"private_state_hashes.data", "private_state_hashes.metadata",
+		"confighistory.data", "confighistory.metadata",
```

Comment:
can't we just use a `expectedSnapshotFiles []string` and append names before each `verifySnapshotOutput()`? I understand that it is a test code only but still it is good to avoid this many numbers of arguments.

Reply:
I predicted this response 😄